podpac/core/data/ogc.py

```diff
@@ -4,373 +4,264 @@
-
-            try:
-                try:  # This works with rasterio v1.0a8 or greater, but not on python 2
-                    with rasterio.open(io) as dataset:
-                        if dotime:
-                            output.data[0, ...] = dataset.read()
-                        else:
-                            output.data[:] = dataset.read()
-                except Exception as e:  # Probably python 2
-                    print(e)
-                    tmppath = os.path.join(settings.cache_path, "wcs_temp.tiff")
-                    if not os.path.exists(os.path.split(tmppath)[0]):
-                        os.makedirs(os.path.split(tmppath)[0])
-                    open(tmppath, "wb").write(content)
-                    with rasterio.open(tmppath) as dataset:
-                        output.data[:] = dataset.read()
-                    os.remove(tmppath)  # Clean up
-            except ImportError:
-                # Writing the data to a temporary tiff and reading it from there is hacky
-                # However reading directly from r.data or io doesn't work
-                # Should improve in the future
-                open("temp.tiff", "wb").write(r.data)
-                try:
-                    output.data[:] = RasterToNumPyArray("temp.tiff")
-                except:
-                    raise Exception("Rasterio or Arcpy not available to read WCS feed.")
-        if not coordinates["lat"].is_descending:
-            if dotime:
-                output.data[:] = output.data[:, ::-1, :]
-            else:
-                output.data[:] = output.data[::-1, :]
+        kwargs = {}
 
-        return output
+        if "time" in coordinates:
+            kwargs["time"] = coordinates["time"].coordinates.astype(str).tolist()
 
-    @property
-    def base_ref(self):
-        """ definition base_ref """
-        if not self.layer_name:
-            return super(WCSBase, self).base_ref
+        if isinstance(self.interpolation, str):
+            kwargs["interpolation"] = self.interpolation
 
-        return self.layer_name.rsplit(".", 1)[1]
+        logger.info(
+            "WCS GetCoverage (source=%s, layer=%s, bbox=%s, shape=%s)"
+            % (self.source, self.layer, (w, n, e, s), (width, height))
+        )
+
+        response = self.client.getCoverage(
+            identifier=self.layer,
+            bbox=(w, n, e, s),
+            width=width,
+            height=height,
+            crs=self.crs,
+            format=self.format,
+            version=self.version,
+            **kwargs
+        )
+        content = response.read()
+
+        # check for errors
+        xml = bs4.BeautifulSoup(content, "lxml")
+        error = xml.find("serviceexception")
+        if error:
+            raise WCSError(error.text)
+
+        # get data using rasterio
+        with rasterio.MemoryFile() as mf:
+            mf.write(content)
+            dataset = mf.open(driver="GTiff")
```

Comment:
Does the driver need to change based on the `self.format`

Reply:
Maybe, I would need an example. Or do you want to throw a `NotImplemetedError("Not yet supported")` if the format is not `geotiff` or `geotiff_byte`, for now?

Reply:
Throw `NotImplementedError` for now.

Reply:
Actually, I just made the trait an enum with only 'geotiff' and 'geotiff_byte'. We can add more later if we need to.